### ISSUE TITLE
Help needed: how to periodically flush logger with appender's immediateFlush set to false?

### ISSUE THREAD

**Issue by c2hpxq** [log4cplus/log4cplus#587]
## TL;DR
I set immediateFlush=false and I want to make sure my log file is up-to-date. So I want to periodically flush it. 

I fail to find a convenient way for that. Any suggestions on that?

## Background
As my program may produce a lot of logs, in consideration of performance, I make the logger's immediateFlush=false to enable buffered IO.
However, sometimes my program may just produce several logs in a quite long period of time. They are buffered thus I can't see them. 
What comes to my mind is that, flush the logger periodically. But after some time of digging into the source code, I fail to find an easy way of achieving that.
The underlying ostream is hided, so I can't operate on it directly;
Logger converts arguments to str and then passes it to ostream, so I can't directly pass std::flush to it.

Any suggestions are welcome :-)

**Comment by wilx** [comment]
There is no periodic flushing of the appenders. You can only adjust the `BufferSize` to be sufficiently small.

**Comment by c2hpxq** [comment]
/8*/*8> There is no periodic flushing of the appenders. You can only adjust the `BufferSize` to be sufficiently small.

Thanks!

I think adjust `BufferSize` + periodically print out 'heart beat' log might help in my situation.
Also I find `DailyRollingFileAppender` might be useful in similar situations.
